npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -476,6 +499,71 @@ def self.dependencies_with_all_versions_metadata(dependency_set)
           dependency
         end
       end
+
+      # Fetch the registry URL for a given package manager
+      sig do
+        params(
+          name: String,
+          lockfiles: T::Hash[Symbol, T.nilable(Dependabot::DependencyFile)],
+          rc_files: T::Hash[Symbol, T.nilable(Dependabot::DependencyFile)]
+        ).returns(String)
+      end
+      def self.find_registry(name, lockfiles, rc_files)
+        find_registry_from_lockfile_or_rc(
+          lockfiles[name.to_sym],
+          rc_files[name.to_sym],
+          default: DEFAULT_REGISTRIES[name]
+        )
+      end
+
+      # Helper to determine registry from lockfile or RC file
+      sig do
+        params(
+          lockfile: T.nilable(Dependabot::DependencyFile),
+          rc_file: T.nilable(Dependabot::DependencyFile),
+          default: String
+        ).returns(String)
+      end
+      def self.find_registry_from_lockfile_or_rc(lockfile, rc_file, default:)
```

Comment:
Any reason to pass the default to this function? Can the caller do that resolution? That way this function only does one thing

Reply:
Because the default for npm, pnpm is different then yarn.